### DISCUSSION TITLE
what's wrong with this function?

### DISCUSSION THREAD

**Discussion by rbhughes** [supabase/supabase#15802] [Questions] [answered]
I copied this known-to-be-okay function from another pg database, but I cannot add it in the supabase UI:
schema = public
return type = trigger
```
CREATE OR REPLACE FUNCTION public.handle_row_changed()
 RETURNS trigger
AS $function$ 
declare
  _oldrow text := md5((to_jsonb(OLD.*) - 'row_changed'::text)::text);
  _newrow text := md5((to_jsonb(NEW.*) - 'row_changed'::text)::text);
begin
	if _newrow <> _oldrow then NEW.row_changed = NOW();
end if;
RETURN NEW;
end;
$function$;
```

I get some variation of this popup error:

`Failed to create function: failed to create pg.functions: syntax error at or near "CREATE"
`

**Comment by GaryAustin1** [comment, marked as answer]
If you are trying to put that in the Dashboard function UI that is a no go.  You should run that in the SQL editor.

If you want to do that in the function UI then you only would use from declare to end;
